src/bootstrap/lib.rs

```diff
@@ -233,6 +234,7 @@ impl Build {
         };
         let rust_info = channel::GitInfo::new(&src);
         let cargo_info = channel::GitInfo::new(&src.join("cargo"));
+        let src_is_git = src.join(".git").is_dir();
```

Comment:
This won't work if rust itself is a git submodule. Checking if .git exists at all would be an improvement.